google/cloud/iam/integration_tests/iam_integration_test.cc

```diff
@@ -226,7 +226,13 @@ TEST_F(IamIntegrationTest, ServiceAccountKeyCrudSuccess) {
       "projects/-/serviceAccounts/" + iam_service_account_,
       {::google::iam::admin::v1::ListServiceAccountKeysRequest::USER_MANAGED});
   ASSERT_STATUS_OK(list_response);
-  EXPECT_GT(list_response->keys().size(), 0);
+  bool found = std::any_of(
+      list_response->keys().begin(), list_response->keys().end(),
+      [&create_response](
+          StatusOr<::google::iam::admin::v1::ServiceAccountKey> const& sa) {
+        return sa.ok() && sa->name() == create_response->name();
+      });
+  EXPECT_TRUE(found);
```

Comment:
nit: a failure here would be really hard to troubleshoot. I often copy the names to a `std::vector<std::string>` and then use `EXPECT_THAT(vector, Contains(expected));`